app/reducers/root-reducer.js

```diff
@@ -20,15 +22,17 @@ export function addToSelectionById(id, state) {
         return true;
     });
 
-    state.selection.unshift(targetItem);
+    selection.unshift(targetItem);
 
-    return Object.assign({}, state);
+    return {...state, collection, selection};
```

Comment:
I feel like an idiot now for not thinking in this way, would you be opposed to formatting the returns as:

```
{
    ...state,
    collection,
    selection
}
```